src/tvheadend/Subscription.cpp

```diff
@@ -264,7 +268,30 @@ void Subscription::ParseSubscriptionStatus ( htsmsg_t *m )
       else if (!strcmp("userLimit", error))
         SetState(SUBSCRIPTION_USERLIMIT);
       else if (!strcmp("noFreeAdapter", error))
-        SetState(SUBSCRIPTION_NOFREEADAPTER);
+      {
+        /* If streaming conflict management enabled */
+        if (Settings::GetInstance().GetStreamingConflict())
+        {
```

Comment:
I would factor out this block to a separate method, e.g. `HandleSubscriptionConflict`.